### PR TITLE
Add new validator for deprecate images

### PR DESCRIPTION
**Pull request by camilamacedo86**
This validator will help us to warning users about the usage of deprecated images found in the CSV

### CODE REVIEW

File: pkg/validation/internal/deprecated_images.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	"strings"
+
+	"github.com/operator-framework/api/pkg/manifests"
+	"github.com/operator-framework/api/pkg/validation/errors"
+	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
+)
+
+// DeprecatedImages contains a list of deprecated images and their respective messages
+var DeprecatedImages = map[string]string{
+	"gcr.io/kubebuilder/kube-rbac-proxy": "Your bundle uses the image `gcr.io/kubebuilder/kube-rbac-proxy`. This upstream image is deprecated and may become unavailable at any point.  \n\nPlease use an equivalent image from a trusted source or update your approach to protect the metrics endpoint.  \n\nFor further information: https://github.com/kubernetes-sigs/kubebuilder/discussions/3907",
```
Comment (camilamacedo86) [inline pkg/validation/internal/deprecated_images.go:14]: It allows us add more images/messages in the future if we need to do so

**Review by perdasilva — APPROVED** [review]
lgtm ^^ thank you!

**Comment by camilamacedo86** [comment]
/hold

I think it will not fly :-( 
Not everybody was onboarding with the idea

**Comment by camilamacedo86** [comment]
closing as not accepted.

**Pull request closed without merging**